public/app/features/explore/state/query.test.ts

```diff
@@ -546,4 +568,187 @@ describe('reducer', () => {
       ).toBeDefined();
     });
   });
+
+  describe('log sample', () => {
+    let dispatch: ThunkDispatch,
+      getState: () => StoreState,
+      unsubscribes: Function[],
+      mockLogsSampleDataProvider: () => Observable<DataQueryResponse>;
+
+    beforeEach(() => {
+      unsubscribes = [];
+      mockLogsSampleDataProvider = () => {
+        return {
+          subscribe: () => {
+            const unsubscribe = jest.fn();
+            unsubscribes.push(unsubscribe);
+            return {
+              unsubscribe,
+            };
+          },
+        } as unknown as Observable<DataQueryResponse>;
+      };
+
+      const store: { dispatch: ThunkDispatch; getState: () => StoreState } = configureStore({
+        ...defaultInitialState,
+        explore: {
+          [ExploreId.left]: {
+            ...defaultInitialState.explore[ExploreId.left],
+            datasourceInstance: {
+              query: jest.fn(),
+              getRef: jest.fn(),
+              meta: {
+                id: 'something',
+              },
+              getLogsSampleDataProvider: () => {
+                return mockLogsSampleDataProvider();
+              },
+            },
+          },
+        },
+      } as unknown as Partial<StoreState>);
+
+      dispatch = store.dispatch;
+      getState = store.getState;
+
+      setupQueryResponse(getState());
+    });
+
+    it('should cancel any unfinished logs sample queries when a new query is run', async () => {
```

Comment:
same as before for all the added tests here, `dispatch` is not async, there's no need to `await` it or mark the test as async